File: src/main/java/net/yakclient/mixin/internal/methodadapter/tree/MethodProxyAdapter.java
package net.yakclient.mixin.internal.methodadapter.tree;

import net.yakclient.YakMixins;
import org.objectweb.asm.MethodVisitor;
import org.objectweb.asm.tree.MethodNode;

public class MethodProxyAdapter extends MethodVisitor {
    private final MethodVisitor parent;

    public MethodProxyAdapter(ASMMethodProxy proxy, MethodVisitor parent) {
        super(YakMixins.ASM_VERSION, proxy);
        this.parent = parent;
    }

    @Override
    public void visitEnd() {
        ((MethodNode) this.mv).accept(this.parent);
    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/methodadapter/tree/MethodProxyAdapter.kt
package net.yakclient.mixins.base.internal.methodadapter.tree

import net.yakclient.mixins.base.internal.bytecode.ByteCodeUtils
import org.objectweb.asm.MethodVisitor
import org.objectweb.asm.tree.MethodNode

class MethodProxyAdapter(
    proxy: TreeMethodProxy,
    private val parent: MethodVisitor
) : MethodVisitor(ByteCodeUtils.ASM_VERSION, proxy) {

    override fun visitEnd() {
        super.visitEnd()
        (mv as MethodNode).accept(parent)
    }
}